dxr/app.py

```diff
@@ -114,27 +122,36 @@ def _search_json(query, tree, query_text, is_case_sensitive, offset, limit, conf
             if is_case_sensitive:
                 params['case'] = 'true'
             return jsonify({'redirect': url_for('.browse', _anchor=line, **params)})
+
     try:
-        count_and_results = query.results(offset, limit)
-        # Convert to dicts for ease of manipulation in JS:
-        results = [{'icon': icon,
-                    'path': path,
-                    'lines': [{'line_number': nb, 'line': l} for nb, l in lines],
-                    'is_binary': is_binary}
-                   for icon, path, lines, is_binary in count_and_results['results']]
+        filters = query.instantiate_filters()
+        # Pull up all the non-negated text terms.
+        text_terms = [term for term in query.terms if term['name'] == 'text' and not term['not']]
```

Comment:
We don't use single_term for promoted paths because the condition is not quite the same: we want promoted paths even when the query is `text path:folder1 ext:abc`, but single_term is for when the query is just `text`.